### PR TITLE
bottom: add missing dependencies when on darwin

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
https://hydra.nixos.org/build/142849916/nixlog/3

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fortuneteller2k** [comment]
ZHF: #122042

**Review by berbiche — APPROVED** [review]

File: pkgs/tools/system/bottom/default.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@ rustPlatform.buildRustPackage rec {
 
   nativeBuildInputs = [ installShellFiles ];
 
-  buildInputs = lib.optional stdenv.hostPlatform.isDarwin darwin.apple_sdk.frameworks.IOKit;
+  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [darwin.apple_sdk.frameworks.IOKit libiconv];
```
Comment (SuperSandro2000) [inline pkgs/tools/system/bottom/default.nix:20]:
```suggestion
  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [ IOKit libiconv ];
```
Darwin shouldn't be used as an input.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 122508` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>bottom</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Here are the logs
```
error: linking with `/nix/store/47vpv5i10dwfg1cf5wca1k40f982g5fm-clang-wrapper-7.1.0/bin/cc` failed: exit
  |
  = note: "/nix/store/47vpv5i10dwfg1cf5wca1k40f982g5fm-clang-wrapper-7.1.0/bin/cc" "-m64" "-arch" "x86_64"
  = note: ld: framework not found DiskArbitration
          clang-7: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error

error: could not compile `sysinfo`
```
Feel free to add it and ping me, I can test again

**Comment by fortuneteller2k** [comment]
> Here are the logs
> 
> ```
> error: linking with `/nix/store/47vpv5i10dwfg1cf5wca1k40f982g5fm-clang-wrapper-7.1.0/bin/cc` failed: exit
>   |
>   = note: "/nix/store/47vpv5i10dwfg1cf5wca1k40f982g5fm-clang-wrapper-7.1.0/bin/cc" "-m64" "-arch" "x86_64"
>   = note: ld: framework not found DiskArbitration
>           clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
> 
> 
> error: aborting due to previous error
> 
> error: could not compile `sysinfo`
> ```
> 
> Feel free to add the missing `DiskArbitration` and ping me, I can test again

Added the missing `DiskArbitration`

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 122508` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>bottom</li>
  </ul>
</details>

**Comment by happysalada** [comment]
here are the logs
```
error: linking with `/nix/store/47vpv5i10dwfg1cf5wca1k40f982g5fm-clang-wrapper-7.1.0/bin/cc` failed: exit
  |
  = note: "/nix/store/47vpv5i10dwfg1cf5wca1k40f982g5fm-clang-wrapper-7.1.0/bin/cc" "-m64" "-arch" "x86_64"
  = note: ld: framework not found Foundation
          clang-7: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error

error: could not compile `sysinfo`
```

sorry that this is so step by step.
Ping me when it's good to test again.
Hopefully that should be the last one.

**Comment by fortuneteller2k** [comment]
> here are the logs
> 
> ```
> error: linking with `/nix/store/47vpv5i10dwfg1cf5wca1k40f982g5fm-clang-wrapper-7.1.0/bin/cc` failed: exit
>   |
>   = note: "/nix/store/47vpv5i10dwfg1cf5wca1k40f982g5fm-clang-wrapper-7.1.0/bin/cc" "-m64" "-arch" "x86_64"
>   = note: ld: framework not found Foundation
>           clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
> 
> 
> error: aborting due to previous error
> 
> error: could not compile `sysinfo`
> ```
> 
> sorry that this is so step by step.
> Ping me when it's good to test again.
> Hopefully that should be the last one.

Added `Foundation`.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 122508` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bottom</li>
  </ul>
</details>

**Comment by happysalada** [comment]
@fortuneteller2k Thanks a lot for this!

**Comment by fortuneteller2k** [comment]
Result of `nixpkgs-review pr 122508` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by happysalada** [comment]
Thanks again!

**Pull request merged**